ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/lvm.py

```diff
@@ -155,157 +192,64 @@ def process(self, device, results, log):
-
-    def lv_state(self, att):
-        # State: (a)ctive, (s)uspended, (I)nvalid snapshot, invalid (S)uspended snapshot,
-        # snapshot (m)erge failed, suspended snapshot (M)erge failed,
-        # mapped (d)evice present without tables, mapped device present with (i)nactive table
-        return {'a': 'active',
-                's': 'suspended',
-                'I': 'invalid snapshot',
-                'S': 'invalid suspended snapshot',
-                'm': 'snapshot merge failed',
-                'M': 'suspended snapshot merge failed',
-                'd': 'mapped device present without tables',
-                'i': 'mapped device present with inactive table'}.get(att, None)
-
-    def lv_device(self, att):
-        # device (o)pen, (X) unknown
-        return {'o': 'open',
-                'X': 'unknown'}.get(att, None)
-
-    def lv_target_type(self, att):
-        # Target  type: (C)ache, (m)irror, (r)aid, (s)napshot, (t)hin, (u)nknown, (v)irtual
-        return {'C': 'cache',
-                'm': 'mirror',
-                'r': 'raid',
-                's': 'snapshot',
-                't': 'thin',
-                'u': 'unknown',
-                'v': 'virtual'}.get(att, None)
-
-    def lv_health(self, att):
-        # Volume Health: (p)artial, (r)efresh needed, (m)ismatches exist, (w)ritemostly, (X) unknown
-        return {'p': 'partial',
-                'r': 'refresh needed',
-                'm': 'mismatches exist',
-                'w': 'writemostly',
-                'X': 'unknown'}.get(att, None)
\ No newline at end of file
+    def makeHDMap(self, columns):
+        hd_om = ObjectMap()
+        hd_om.title = columns['disk']
+        hd_om.id = 'disk-{}'.format(self.prepId(columns['disk']))
+        hd_om.size = int(columns['size'])
+        hd_om.relname = 'harddisks'
+        hd_om.modname = 'ZenPacks.zenoss.LinuxMonitor.HardDisk'
+        return hd_om
+
+    def makePVMap(self, columns):
+        # pv_name,pv_fmt,pv_attr,pv_size,pv_free,pv_uuid,vg_name
+        pv_om = ObjectMap()
+        pv_om.title = columns['pv_name']
+        pv_om.id = 'pv-{}'.format(self.prepId(columns['pv_name']))
+        pv_om.format = columns['pv_fmt']
+        pv_om.attributes = self.lvm_parser.pv_attributes(columns['pv_attr'])
+        pv_om.pvsize = int(columns['pv_size'])
+        pv_om.free = int(columns['pv_free'])
+        pv_om.uuid = columns['pv_uuid']
+        pv_om.set_volumeGroup = 'vg-{}'.format(columns['vg_name'])
+        pv_om.relname = 'physicalVolumes'
+        pv_om.modname = 'ZenPacks.zenoss.LinuxMonitor.PhysicalVolume'
+        return pv_om
+
+    def makeVGMap(self, columns):
+        # vg_name,vg_attr,vg_size,vg_free,vg_uuid
+        vg_om = ObjectMap()
+        vg_om.title = columns['vg_name']
+        vg_om.id = 'vg-{}'.format(self.prepId(columns['vg_name']))
+        vg_om.attributes = self.lvm_parser.vg_attributes(columns['vg_attr'])
+        vg_om.vgsize = int(columns['vg_size'])
```

Comment:
Setting VolumeGroup.vgsize is resulting in the following zenhub.log warning during modeling. There does appear to be a vgsize property defined on VolumeGroup in zenpack.yaml. I think the problem is that the property has `datapoint` set, which causes zenpacklib to make it a method instead of a property. It looks like you need to choose whether this is something that will be modeled or monitored. If it'll be modeled you should remove the `datapoint` field from the property in zenpack.yaml, or if it'll be monitored you need to stop setting vg_om.vgsize during modeling.

```
2016-02-24 19:00:46,964 INFO zen.ZenHub: Worker (307) reports 2016-02-24 19:00:46,964 WARNING zen.ApplyDataMap: getter for 'vgsize' not found on obj 'vg-centos', skipping
```